File: factory/go/dotaHeroFactory.go
package main

import "fmt"

func getHero(heroType string) (iHero, error) {
	if heroType == "Sniper" {
		return newSniper(), nil
	} else if heroType == "Axe" {
		return newAxe(), nil
	}
	return nil, fmt.Errorf("Hero not found")
}

File: java/src/DotaHeroFactory.java
public class DotaHeroFactory {

  Hero getHero(String type) {
    if (type.equals("Sniper")) {
      return new Sniper();
    } else if (type.equals("Axe")) {
      return new Axe();
    } else {
      return null;
    }
  }
}
